### PR TITLE
Add tests to terminal service

### PR DESCRIPTION
**Pull request by hexa00**
This also adds the yarn test:slow command to do integration tests.
It is only enabled for the terminal package at the moment

### CODE REVIEW

File: packages/terminal/src/node/terminal-backend-contribution.slow-spec.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+chai.use(chaiAsPromised);
+
+/**
+ * Globals
+ */
+
+const expect = chai.expect;
+
+describe('Termninal Backend Contribution', function () {
```
Comment (svenefftinge) [inline packages/terminal/src/node/terminal-backend-contribution.slow-spec.ts]: typo 'Terminal'

File: packages/terminal/src/node/inversify.spec-config.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+/*
```
Comment (svenefftinge) [inline packages/terminal/src/node/inversify.spec-config.ts]: According to https://github.com/theia-ide/theia/blob/master/doc/Testing.md#test-directory-structure this should go to `packages/terminal/src/node/test/inversify.spec-config.ts`

Comment (hexa00) [reply packages/terminal/src/node/inversify.spec-config.ts]: Right, fixing

File: packages/terminal/src/node/terminal-backend-contribution.slow-spec.ts (hunk 1 above)
Comment (hexa00) [reply packages/terminal/src/node/terminal-backend-contribution.slow-spec.ts]: oops

File: packages/terminal/extension.package.json [hunk 3]
```diff
@@ -10,6 +10,9 @@
     "@types/xterm": "^2.0.3",
     "xterm": "^2.8.1"
   },
+  "devDependencies": {
```
Comment (akosyakov) [inline packages/terminal/extension.package.json]: Do you really need it? ws should be available via @theia/core

Comment (hexa00) [reply packages/terminal/extension.package.json]: Don't I need to specific direct dependencies ?

Comment (akosyakov) [reply packages/terminal/extension.package.json]: all transitive are available automatically, with the redeclaration we have to manage versions in 2 places

Comment (hexa00) [reply packages/terminal/extension.package.json]: OK I guess I have to get used to this... it's not easy with the hoisting you have to manually verify that one of your direct deps includes it... testing it is not enought

**Comment by hexa00** [comment]
I think so yes unless it becomes an issue to run them.

**Pull request merged**